shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -903,6 +903,12 @@ - (UITextContentType)textContentType {
 // Prevent UIKit from showing selection handles or highlights. This is needed
 // because Scribble interactions require the view to have it's actual frame on
 // the screen.
+//
+// These are not documened methods. On iOS 17, the insertion point color is also
+// used to paint the highlighted background of the selected IME candidate:
+// https://github.com/flutter/flutter/issues/132548
+// So the respondsToSelector method is overridden to return NO for this method
+// on iOS 17+.
```

Comment:
I'm confused as to why we are both implementing it but claiming not to. Are we relying on an implementation detail of some system code calling this unconditionally and other code checking whether its implemented first, and relying on that distinction to make things work? If so, that seems incredibly fragile, and the sort of thing that's likely to cause another round of fire drills in a subsequent iOS version.

Reply:
UIKit falls back to `UIColor.insertionPointColor` if the receiver doesn't respond to `insertionPointColor`. That color is unfortunately private. 

> some system code calling this unconditionally

The method is actually private (but it's in the codebase for a while and no complains from app store) and it's not part of any protocols, so UIKit has to check `respondsToSelector` first.

Reply:
> The method is actually private (but it's in the codebase for a while and no complains from app store) and it's not part of any protocols, so UIKit has to check `respondsToSelector` first.

Then how does this change not break the "Prevent UIKit from showing selection handles or highlights" use case that the implementation is documented as being for?

Reply:
On iOS 17 there's new API for adding system UI for text selection/cursor/highlights: https://developer.apple.com/documentation/uikit/uitextselectiondisplayinteraction?language=objc . Without adding that interaction to the input view (and we don't add  [`UIITextInteraction`](https://developer.apple.com/documentation/uikit/uitextinteraction?language=objc) to the view unless the framework is currently sending editing state updates, as a hack to fix a different bug), UIKit doesn't seem to add selection handles or highlights automatically on iOS 17.

Reply:
I see; please update the comments on this method then to explain this, and that it's bypassed on iOS 17+ by a `respondsToSelector:` check.

Reply:
Hi @fbcouch do you remember when is `insertionPointColor` called during scribble? I'd like to verify if this is still needed on iOS17.

Reply:
If I recall correctly, it was basically during any text interaction – since we changed to show the frame in it's proper place on the screen, iPadOS was able to draw it's own selection handles, etc